### PR TITLE
docs: Update the poetry installation documentation

### LINKED ISSUE #376: Poetry has changed their installer
**Description**:
Poetry 1.2 has deprecated the existing installer that is used in the readme, this installer is also recommended for previous versions. The instructions need to be changed.
See the issue poetry has about the installer being deprecated: https://github.com/python-poetry/poetry/issues/6377

**Acceptance criteria**:
- [x] Document new installation process
  - [x] New installation script
  - [x] New installation location

### PR DESCRIPTION
**Pull request by Reillyhewitson**
This PR will close #376

## Description
Enter a description of the changes here

## Testing Instructions
Add a set up instructions describing how the reviewer should test the code

- [x] Review code
- [ ] Check GitHub Actions build
- [x] If `icatdb Generator Script Consistency Test` CI job fails, is this because of a deliberate change made to the script to change generated data (which isn't actually a problem) or is here an underlying issue with the changes made?
- [x] Review changes to test coverage
- [x] Does this change mean a new patch, minor or major version should be made? If so, does one of the commit messages feature `fix:`, `feat:` or `BREAKING CHANGE:` so a release is automatically made via GitHub Actions upon merge?
- [ ] {more steps here}

## Agile Board Tracking
Connect to #376

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -103,15 +103,26 @@ To maintain records of the API's dependencies,
...
-curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python -
+curl -sSL https://install.python-poetry.org | python3 -
 ```
 
-The installation requires `~/.poetry/env` to be refreshed for changes to be applied.
-Open a new terminal or execute the following command to ensure the installation is
-completed smoothly:
+The installation requires the following to be added to your `~/.bashrc` file so the installation folder is on your path.
+```bash
+export PATH= "~/.local/bin:$PATH"
```
Comment (MRichards99) [inline README.md:111]: There is a space between `=` and `"` that shouldn't be there. Could this be removed?

File: README.md [hunk 2]
```diff
@@ -103,15 +103,26 @@ To maintain records of the API's dependencies,
...
+```
+Then run `source ~/.bashrc` or open a new terminal and check poetry works by running `poetry --version`
 
+if you encounter this error when installing poetry:
```
Comment (MRichards99) [inline README.md:115]: The i in "if" should be capitalised

**Review by MRichards99 — CHANGES_REQUESTED** [review]
Looks good, just a couple of minor changes to make. Is the dependency safety failure due to `dparse`? The logs have disappeared for me but you might be able to remember? If you can't just re-run the CI again so we can see.

**Comment by Reillyhewitson** [comment]
> Looks good, just a couple of minor changes to make. Is the dependency safety failure due to `dparse`? The logs have disappeared for me but you might be able to remember? If you can't just re-run the CI again so we can see.

Yes, the dependency failure was due to dparse. I'll change the things you've noted and rerun the CI.

**Comment by MRichards99** [comment]
@Reillyhewitson Yes re-run the CI just to confirm. When the PR is merged, the failure will cause the CI to fail on the main branch so we might delay merging the PR until we can fix dparse/Poetry 1.2

**Comment by Reillyhewitson** [comment]
Just reran the CI and can confirm that the problem is dparse

**Comment by MRichards99** [comment]
Are you going to update dparse as part of #372? I forgot yesterday that you might be able to update that dependency using Poetry 1.1.9

File: README.md (hunk 1 above)
Comment (Reillyhewitson) [reply README.md]: Has been removed

File: README.md (hunk 2 above)
Comment (Reillyhewitson) [reply README.md]: If has been capitalised

File: README.md [hunk 3]
```diff
@@ -103,15 +103,26 @@ To maintain records of the API's dependencies,
 command:
 
 ```bash
-curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python -
+curl -sSL https://install.python-poetry.org | python3 -
```
[the remark this answers is not in the document]
Comment (Reillyhewitson) [reply README.md:106]: Updated to the new poetry installation URL.

**Review by MRichards99 — APPROVED** [review]

**Comment by Reillyhewitson** [comment]
Merge after https://github.com/ral-facilities/datagateway-api/pull/381 so that pydantic doesn't cause dparse to fail

**Pull request merged**